lib/elixir/test/elixir/option_parser_test.exs

```diff
@@ -400,3 +288,188 @@ defmodule OptionParserTest do
     assert original == OptionParser.to_argv(opts, switches: [counter: :count])
   end
 end
+
+defmodule OptionsParserDeprecationsTest do
+  use ExUnit.Case, async: false
+
+  @warning ~r[Not passing the :switches or :strict option to OptionParser is deprecated]
+
+  defmacro assert_deprecated(clause) do
```

Comment:
I am not convinced we need a macro here. Could we get away with:

```elixir
defp assert_deprecated(fun) do
  assert ExUnit.CaptureIO.capture_io(:stderr, fun) =~ @warning
end
```

and

```elixir
assert_deprecated(fn ->
  assert OptionParser.parse(["--docs"]) == {[docs: true}, [], []}
end)
```

?